src/renderer/MicrophoneSoundBar.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { SettingsContext } from './App';
+
+const TestMicrophoneButton: React.FC = () => {
+    const [{ microphone }] = useContext(SettingsContext)
+    const [error, setError] = useState<boolean>(false)
+    const [rms, setRms] = useState<number>(0)
+
+    useEffect(() => {
+        setError(false)
+
+        const ctx = new AudioContext()
+        const processor = ctx.createScriptProcessor(2048, 1, 1)
+        processor.connect(ctx.destination)
+
+        const handleProcess = (event: AudioProcessingEvent) => {
+            if (Math.floor(event.timeStamp / 10) % 50 !== 0) return
```

Comment:
This doesn't work as expected due to potential randomness in update frequency.

here is a better implementation: 

```js
        const minUpdateRate = 50;
        let lastRefreshTime = 0;

        const handleProcess = (event: AudioProcessingEvent) => {
            // limit update frequency
            if ( event.timeStamp - lastRefreshTime < minUpdateRate ) {
              return;
            }

            // update last refresh time
            lastRefreshTime = event.timeStamp;

            const input = event.inputBuffer.getChannelData(0)
            const total = input.reduce((acc, val) => acc + Math.abs(val), 0)
            const rms = Math.sqrt(total / input.length)
            setRms(rms)
        }
```

This makes sure a *minimum* of 50ms occurs between updates by comparing the difference in time instead of relying on timestamp values occurring at a regular and reliable interval. 

![crew-link-mic-test-lag-fix](https://user-images.githubusercontent.com/45262335/101024801-78b52200-3529-11eb-9755-944777ffbca3.gif)

Reply:
Oh wow, that does work a lot better, thanks! I didn't think about having a local variable in the `useEffect`. I've always been pretty religious to throw all state into `useContext`, `useState`, or `useReducer`. But this works great!